src/harness/harness.ts

```diff
@@ -959,6 +961,9 @@ module Harness {
                 options.module = options.module || ts.ModuleKind.None;
                 options.newLine = options.newLine || ts.NewLineKind.CarriageReturnLineFeed;
                 options.noErrorTruncation = true;
+                if (lightMode) {

```

Comment:
have light mode be a local here.  Possibly default it to 'true'.  Then check test settings (which might set it to false if that's what the test does).   Then, from that point on you can use the local.